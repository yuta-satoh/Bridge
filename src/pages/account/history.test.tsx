import renderer from 'react-test-renderer';
import { Middleware } from 'swr';
import { SWRResponse } from 'swr';
import { SWRConfig } from 'swr';
import History from './history';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';


type Cart = {
    id: number;
    item_id: number;
    user_id: number;
    name: string;
    description: string;
    genre: string;
    category: string;
    price: number;
    imgurl: string;
    date: string;
    quantity: number;
    delete: boolean;
  }[];

jest.mock('next/router', () => ({
  useRouter() {
    return {
      push: jest.fn(),
    };
  },
}));

test('描画される', async() => {
  const historyData = [
    {
      category: 'テーブル',
      date: '2023-03-01',
      delete: false,
      description: 'すっきりと洗練されたデザインのテーブル。',
      genre: 'フェミニン',
      id: 4,
      imgurl: '/images/table/table_feminine_1.jpg',
      item_id: 4,
      name: 'ガラスローテーブル',
      price: 29000,
      quantity: 1,
      user_id: 1,
    },
  ];
  const testMiddleware: Middleware = () => {
    return (): SWRResponse<any, any> => {
      return {
        data: historyData,
        error: undefined,
        mutate: (_) => Promise.resolve(),
        isValidating: false,
        isLoading: false,
      };
    };
  };
  const tree = renderer
    .create(
      <SWRConfig value={{ use: [testMiddleware] }}>
        <History cookie={'1'} />
      </SWRConfig>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('履歴はありませんが表示', async() => {
    const historyData:Cart = [];
    const testMiddleware: Middleware = () => {
      return (): SWRResponse<any, any> => {
        return {
          data: historyData,
          error: undefined,
          mutate: (_) => Promise.resolve(),
          isValidating: false,
          isLoading: false,
        };
      };
    };
    const tree = renderer
      .create(
        <SWRConfig value={{ use: [testMiddleware] }}>
          <History cookie={'1'} />
        </SWRConfig>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
