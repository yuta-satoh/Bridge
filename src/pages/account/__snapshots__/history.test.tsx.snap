// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`履歴はありませんが表示 1`] = `
[
  <div
    className="breadList"
  >
    <ol
      className="links"
      id="top"
    >
      <li
        className="pageLink"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Bridge
        </a>
        <span
          className="greaterThan"
        >
          &gt;
        </span>
      </li>
      <li
        className="pageLink"
      >
        <a
          href="/mypage"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          マイページ
        </a>
        <span
          className="greaterThan"
        >
          &gt;
        </span>
      </li>
      <li
        className="pageLink"
      >
        購入履歴
      </li>
    </ol>
  </div>,
  <div
    className="body"
  >
    <h1
      className="noData"
    >
      履歴はありません
    </h1>
    <div
      className="loginLink"
    >
      <a
        href="/mypage"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onTouchStart={[Function]}
      >
        <button
          className="linkButton"
          type="button"
        >
          マイページ
          <span
            className="buttonSpan"
          >
            →
          </span>
        </button>
      </a>
    </div>
  </div>,
]
`;

exports[`描画される 1`] = `
[
  <div
    className="breadList"
  >
    <ol
      className="links"
      id="top"
    >
      <li
        className="pageLink"
      >
        <a
          href="/"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          Bridge
        </a>
        <span
          className="greaterThan"
        >
          &gt;
        </span>
      </li>
      <li
        className="pageLink"
      >
        <a
          href="/mypage"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onTouchStart={[Function]}
        >
          マイページ
        </a>
        <span
          className="greaterThan"
        >
          &gt;
        </span>
      </li>
      <li
        className="pageLink"
      >
        購入履歴
      </li>
    </ol>
  </div>,
  <div
    className="body"
  >
    <h1
      className="title"
    >
      購入履歴
    </h1>
    <div
      className="historyOrder"
    >
      <label
        htmlFor="historyOrder"
      >
        並び替える: 
      </label>
      <label
        className="jsx-2982400643 selectbox"
      >
        <select
          className="jsx-2982400643"
          data-testid="select-option"
          id="historyOrder"
          name="historyOrder"
          onChange={[Function]}
          value="新しい順"
        >
          <option
            className="jsx-2982400643"
            value="新しい順"
          >
            新しい順
          </option>
          <option
            className="jsx-2982400643"
            value="古い順"
          >
            古い順
          </option>
        </select>
      </label>
    </div>
    <table
      className="tableBody"
    >
      <thead>
        <tr>
          <th />
          <th>
            商品名
          </th>
          <th
            className="tableCell"
          >
            購入数
          </th>
          <th
            className="tableCell"
          >
            単価
          </th>
          <th
            className="tableCell"
          >
            小計
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          className="tableLine"
        >
          <td
            className="tableCellCenter"
          >
            <a
              href="../items/itemlist/4"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              <img
                alt="ガラスローテーブル"
                className="cardImage"
                data-nimg="1"
                decoding="async"
                height={100}
                loading="lazy"
                onError={[Function]}
                onLoad={[Function]}
                src="/_next/image?url=%2Fimages%2Ftable%2Ftable_feminine_1.jpg&w=256&q=75"
                srcSet="/_next/image?url=%2Fimages%2Ftable%2Ftable_feminine_1.jpg&w=128&q=75 1x, /_next/image?url=%2Fimages%2Ftable%2Ftable_feminine_1.jpg&w=256&q=75 2x"
                style={
                  {
                    "color": "transparent",
                  }
                }
                width={100}
              />
            </a>
          </td>
          <td
            className="tableCellCenter"
          >
            <a
              href="../items/itemlist/4"
              onClick={[Function]}
              onMouseEnter={[Function]}
              onTouchStart={[Function]}
            >
              ガラスローテーブル
            </a>
          </td>
          <td
            className="tableCell"
          >
            1
            個
          </td>
          <td
            className="tableCell"
          >
            ¥ 
            31,900
          </td>
          <td
            className="tableCell"
          >
            ¥
             
            31,900
          </td>
          <td
            className="tableCellCenter"
          >
            <button
              className="buttonStyle"
              onClick={[Function]}
              type="button"
            >
              レビューする
            </button>
          </td>
        </tr>
        <tr>
          <td
            colSpan={5}
          />
          <td
            className="tableCellCenterSub"
          >
            購入日：
            2023-03-01
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
